tasks/libs/ciproviders/gitlab_api.py

```diff
@@ -180,6 +192,8 @@ def apply_yaml_extends(config: dict, node):
             for key, value in parent.items():
                 if key not in node:
                     node[key] = value
+                if key in node and isinstance(node[key], dict) and isinstance(value, dict):
```

Comment:
I think you should use `elif` since `key` will always be in `node` here.
```suggestion
                elif key in node and isinstance(node[key], dict) and isinstance(value, dict):
```

Reply:
Updated with `elif`.
I do not think we can early return because we still need to process the rest of items.
I could use `continue` but that would not change anything I think